runtime/include/module.h

```diff
@@ -62,6 +88,14 @@ struct module {
 	struct admissions_info admissions_info;
 	uint64_t               relative_deadline; /* cycles */
 
+	/* Deferrable Server Attributes */
+	uint64_t                 replenishment_period; /* cycles, not changing after init */
+	uint64_t                 max_budget;           /* cycles, not changing after init */
+	_Atomic volatile int64_t remaining_budget;     /* cycles left till next replenishment, can be negative */
+
+	struct perworker_module_sandbox_queue *pwm_sandboxes;
```

Comment:
No idea what pwm stands for. Normally it is pulse-width modulated...

Reply:
It is an acronym following the struct name "per worker module". It is ugly, though, I know.